main/src/ui/unified_window_controller.vala

```diff
@@ -142,6 +142,15 @@ public class UnifiedWindowController : Object {
         }
     }
 
+    public void loop_conversations(bool backwards = false) {
+        Gee.List<Conversation> conversations = stream_interactor.get_module(ConversationManager.IDENTITY).get_active_conversations();
+        conversations.sort((a, b) => { return b.last_active.compare(a.last_active); });
+        int index = conversations.index_of(this.conversation);
+        index += backwards ? -1 : 1;
+        select_conversation(conversations.get(index % conversations.size));
```

Comment:
-1 % n != n
It wouldn't go to the last conversation when pressing Ctrl+Shift+Tab on the first conversation.